File: wayland/src/main/java/org/westmalle/wayland/core/UnsupportedBuffer.java
//Copyright 2016 Erik De Rijcke
//
//Licensed under the Apache License,Version2.0(the"License");
//you may not use this file except in compliance with the License.
//You may obtain a copy of the License at
//
//http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,software
//distributed under the License is distributed on an"AS IS"BASIS,
//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express or implied.
//See the License for the specific language governing permissions and
//limitations under the License.
package org.westmalle.wayland.core;

import com.google.auto.value.AutoValue;
import org.freedesktop.wayland.server.WlBufferResource;

import javax.annotation.Nonnegative;
import javax.annotation.Nonnull;

@AutoValue
public abstract class UnsupportedBuffer implements Buffer {

    public static UnsupportedBuffer create(@Nonnull WlBufferResource wlBufferResource) {
        return new AutoValue_UnsupportedBuffer(wlBufferResource);
    }

    @Nonnegative
    @Override
    public int getWidth() {
        return 0;
    }

    @Nonnegative
    @Override
    public int getHeight() {
        return 0;
    }

    @Nonnull
    @Override
    public abstract WlBufferResource getWlBufferResource();

    @Override
    public void accept(@Nonnegative final BufferVisitor bufferVisitor) {
        bufferVisitor.visit(this);
    }
}

File: compositor/src/main/kotlin/org/westford/compositor/core/UnsupportedBuffer.kt
/*
 * Westford Wayland Compositor.
 * Copyright (C) 2016  Erik De Rijcke
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.westford.compositor.core

import org.freedesktop.wayland.server.WlBufferResource

import javax.annotation.Nonnegative

data class UnsupportedBuffer(@param:Nonnegative override val width: Int,
                             @param:Nonnegative override val height: Int,
                             override val wlBufferResource: WlBufferResource) : Buffer {

    override fun accept(bufferVisitor: BufferVisitor) = bufferVisitor.visit(this)
}
